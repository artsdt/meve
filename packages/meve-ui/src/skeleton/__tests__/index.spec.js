import Skeleton from '..'
import { mount } from '@vue/test-utils'

test('test skeleton loading pass true', async () => {
  const wrapper = mount(Skeleton, {
    propsData: {
      loading: true,
    },
    scopedSlots: {
      default: '<div class="skeleton-default-slot"></div>',
    },
  })

  expect(wrapper.find('.skeleton-default-slot').exists()).toBeFalsy()

  wrapper.destroy()
})

test('test skeleton loading pass false', async () => {
  const wrapper = mount(Skeleton, {
    propsData: {
      loading: false,
    },
    scopedSlots: {
      default: '<div class="skeleton-default-slot"></div>',
    },
  })

  expect(wrapper.find('.skeleton-default-slot').exists()).toBeTruthy()
  wrapper.destroy()
})

test('test skeleton styles', () => {
  const wrapper = mount(Skeleton, {
    propsData: {
      title: true,
      card: true,
      avatar: true,
      titleWidth: '100px',
      avatarSize: '100px',
      cardHeight: '100px',
      rows: 2,
      rowsWidth: ['600px', '500px'],
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test skeleton fullscreen', () => {
  const wrapper = mount(Skeleton, {
    propsData: {
      fullscreen: true,
      fullscreenZIndex: 999,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
