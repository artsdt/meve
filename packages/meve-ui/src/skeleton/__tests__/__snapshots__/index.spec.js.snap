// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test skeleton fullscreen 1`] = `
"<div class=\\"m-skeleton m--box\\">
  <div class=\\"m-skeleton__fullscreen\\" style=\\"z-index: 999;\\">
    <div class=\\"m-skeleton--animation\\"></div>
  </div>
</div>"
`;

exports[`test skeleton styles 1`] = `
"<div class=\\"m-skeleton m--box\\">
  <div class=\\"m-skeleton__content\\">
    <div class=\\"m-skeleton__card\\" style=\\"height: 100px;\\">
      <div class=\\"m-skeleton--animation\\"></div>
    </div>
    <div class=\\"m-skeleton__article\\">
      <div class=\\"m-skeleton__avatar\\" style=\\"width: 100px; height: 100px;\\">
        <div class=\\"m-skeleton--animation\\"></div>
      </div>
      <div class=\\"m-skeleton__section\\">
        <div class=\\"m-skeleton__title\\" style=\\"width: 100px;\\">
          <div class=\\"m-skeleton--animation\\"></div>
        </div>
        <div class=\\"m-skeleton__row\\" style=\\"width: 600px;\\">
          <div class=\\"m-skeleton--animation\\"></div>
        </div>
        <div class=\\"m-skeleton__row\\" style=\\"width: 500px;\\">
          <div class=\\"m-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
